config/routes.rb

```diff
@@ -123,6 +123,10 @@
       :as => :edit_cart_item
   # delete '/cart/empty' => 'application#empty_cart', :as => :empty_cart
   # put '/cart/update' => 'application#update_cart', :as => :update_cart
+  post '/edit_reservation_form' => 'catalog#submit_cart_updates_form',
+       :as => :submit_cart_updates_form
+  put '/change_reservation_dates' => 'catalog#change_reservation_dates',
+      :as => :change_reservation_dates
 
```

Comment:
@AlanLiu96 - You're using the <%= form_tag url_for(action: 'reload_catalog_cart', controller: "catalog").... and the the match at the end of the routs will catch it, there's no need to specify these routs...